src/main/java/htsjdk/samtools/cram/build/CompressionHeaderFactory.java

```diff
@@ -46,103 +39,101 @@
 import java.util.TreeMap;
 
 /**
- * A class responsible for decisions about which encodings to use for a given set of records.
- * This particular version relies heavily on GZIP and RANS for better compression.
+ * Factory for creating CRAM compression headers for containers when writing to a CRAM stream. The fixed data
+ * series are generally the same for every container in the stream (this is not reuired by the spec, but reflects
+ * the current htsjsk implmentation), however, the tag data series and encodings can vary across containers
+ * based on which tags are present in the actual records for that container, and the best compressor to use based
+ * on the actual data. This class delegates to a {@link CRAMEncodingStrategy} object to determine which encodings
+ * to use for the fixed CRAM data series, and dynamically chooses the best encoding for tag data series.
  */
 public class CompressionHeaderFactory {
     public static final int BYTE_SPACE_SIZE = 256;
     public static final int ALL_BYTES_USED = -1;
 
```

Comment:
The refactoring in this class seems pretty innocuous to me and I don't see any issues that need addressing.